tests/storage/storage_backend.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Generasl Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+#
+# Refer to the README and COPYING files for full details of the license
+#
+
+import urllib.parse
+
+import pytest
+
+
+class Backend:
+    """
+    Wrap a userstorage.Backend, adding a url and context manager interface to
+    simplify fixtures.
+    """
+
+    def __init__(self, storage, can_detect_sector_size=True):
+        if not storage.exists():
+            pytest.xfail(f"Storage {storage.name} is not available")
+
+        self._storage = storage
+        self.path = storage.path
+        self.url = urllib.parse.urlparse(f"file:{storage.path}")
+        self.block_size = storage.sector_size
+        self.can_detect_sector_size = can_detect_sector_size
+
+    def __enter__(self):
+        self._storage.setup()
+        return self
+
+    def __exit__(self, *args):
+        self._storage.teardown()
```

Comment:
We can add here:

    @contextmanager
    def tmpdir(self):
        yield tempfile.TemporaryDirectory(dir=self.path)

Of course this works only with "mount" user storage, so it may need to be smarter.

Reply:
Sounds good. But as is it does not work. This yields a `TemporaryDirectory` instance, therefore we cannot do this:
```python
with Backend(storage) as backend:
    yield backend.tmpdir()
```
As it just gives the instance. We want to enter the context and get the directory path, so we need to do:
```python
with Backend(storage) as backend:
    with backend.tmpdir() as tmpdir:
        return tmpdir
```
But in this case it does not execute the TemporaryDirectory `__enter__` method, since we wrapped the context manager class with the `contextmanager` decorator. So it is not what we want either: we do not get the directory name, but the class instance; and it will not be cleaned up at exit.

Instead, I just return the class without the `@contextmanager` decorator.

Reply:
Sounds good.